codex-rs/core/src/shell.rs

```
@@ -0,0 +1,125 @@
+use shlex;
+use std::process::Command;
+use whoami;
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum Shell {
+    Zsh(String),
+    Unknown,
+}
+
+impl Shell {
+    pub fn run_with_profile(&self, command: Vec<String>) -> Option<Vec<String>> {
+        match self {
+            Shell::Zsh(shell_path) => {
+                let mut result = vec![shell_path.clone(), "-c".to_string()];
+                if let Ok(joined) = shlex::try_join(command.iter().map(|s| s.as_str())) {
+                    result.push(format!("source ~/.zshrc && ({joined})"));
```

Is `~/.zshrc` required to exist? Should `;` be used instead of `&&` to ignore a failure in that case?

---

A well-behaved `~/.zshrc` should not do this, but it could write to stdout/stderr and interfere with the tool call output, no?

---

Added logic to check for zshrc